zone/parcels.cpp

```diff
@@ -0,0 +1,725 @@
+							Chat::Yellow,
+							PARCEL_INV_FULL,
+							merchant->GetCleanName()
+						);
+						SendParcelRetrieveAck();
+						return;
+					}
+				}
+				else {
+					MessageString(
+						Chat::Yellow,
+						PARCEL_INV_FULL,
+						merchant->GetCleanName()
+					);
+					SendParcelRetrieveAck();
+					return;
+				}
+			}
+		}
+
+		if (player_event_logs.IsEventEnabled(PlayerEvent::PARCEL_RETRIEVE)) {
+			PlayerEvent::ParcelRetrieve e{};
+			e.from_player_name = p->second.from_name;
+			e.item_id          = p->second.item_id;
+			e.quantity         = p->second.quantity;
+			e.sent_date        = p->second.sent_date;
+
+			RecordPlayerEventLog(PlayerEvent::PARCEL_RETRIEVE, e);
+		}
+
+		DeleteParcel(p->second.id);
+		SendParcelDelete(parcel_in);
+		parcels.erase(p);
+	}
+	SendParcelRetrieveAck();
+	SendParcelIconStatus();
+}
+
+bool Client::DeleteParcel(uint32 parcel_id)
+{
+	auto result = BaseParcelsRepository::DeleteOne(database, parcel_id);
+	if (!result) {
+		LogError("Error deleting parcel id {} from the database.", parcel_id);
+		return false;
+	}
+
+	auto it = std::find_if(parcels.cbegin(), parcels.cend(), [&](const auto &x) { return x.second.id == parcel_id; });
+	SetParcelCount(GetParcelCount() - 1);
+
+	return true;
+}
+
+void Client::LoadParcels()
+{
+	parcels.clear();
+	auto results = ParcelsRepository::GetWhere(database, fmt::format("to_name = '{}'", GetCleanName()));
+
+	for (auto const &p: results) {
+		parcels.emplace(p.slot_id, p);
+	}
+
+	SetParcelCount(parcels.size());
+}
+
+void Client::SendParcelDelete(const ParcelRetrieve_Struct parcel_in)
+{
+	auto outapp = new EQApplicationPacket(OP_ShopDeleteParcel, sizeof(ParcelRetrieve_Struct));
+	auto data   = (ParcelRetrieve_Struct *) outapp->pBuffer;
+
+	data->merchant_entity_id = parcel_in.merchant_entity_id;
+	data->player_entity_id   = parcel_in.player_entity_id;
+	data->parcel_slot_id     = parcel_in.parcel_slot_id;
+	data->parcel_item_id     = parcel_in.parcel_item_id;
+
+	FastQueuePacket(&outapp);
+}
+
+
+int32 Client::FindNextFreeParcelSlot(std::string &character_name)
+{
+	auto results = ParcelsRepository::GetWhere(
+		database, fmt::format("to_name = '{}' ORDER BY slot_id ASC", character_name.c_str()));
+
+	if (results.empty()) {
+		return PARCEL_BEGIN_SLOT;
+	}
+	else {
+		for (uint32 i = PARCEL_BEGIN_SLOT; i <= PARCEL_END_SLOT; i++) {
+			auto it = std::find_if(results.cbegin(), results.cend(), [&](const auto &x) { return x.slot_id == i; });
+			if (it != results.end()) {
+				continue;
+			}
+			else {
+				return i;
+			}
+		}
+
+		return INVALID_INDEX;
+	}
+}
+
+void Client::SendParcelIconStatus()
+{
+	auto outapp = new EQApplicationPacket(OP_ShopParcelIcon, sizeof(ParcelIcon_Struct));
+	auto data   = (ParcelIcon_Struct *) outapp->pBuffer;
+
+	auto const num_of_parcels = GetParcelCount();
+
+	if (num_of_parcels == 0) {
+		data->status = IconOff;
+	}
+	else if (num_of_parcels > PARCEL_MAX_ITEMS) {
+		data->status = Overlimit;
+	}
+	else {
+		data->status = IconOn;
+	}
```

Comment:
The default case can be on top and remove the else

```cpp
data->status = IconOn;
if (num_of_parcels == 0) {
	data->status = IconOff;
}
else if (num_of_parcels > PARCEL_MAX_ITEMS) {
	data->status = Overlimit;
}
```

Reply:
Resolved